### ISSUE TITLE
TimeMOC resolution issue

### ISSUE THREAD

**Issue by BaptisteCecconi** [cds-astro/mocpy#58]
The Time objects of astropy.time are coded with 2 double-precision floating point numbers. This allows to have access to microsecond resolution.

Since mocpy is using astropy.time, we should have access to microsecond resolution (the deepest TimeMOC order), but we don't.

Simple code to test this:
```
>>> import mocpy
>>> from astropy.time import Time
>>> tmin = Time(['2000-01-01T00:00:00.000000Z','2000-01-01T00:00:00.000010Z'])
>>> tmax = Time(['2000-01-01T00:00:00.000005Z','2000-01-01T00:00:00.000015Z'])
>>> (tmax-tmin).sec
array([5.00000041e-06, 5.00000041e-06])
>>> tm = mocpy.TimeMOC.from_time_ranges(tmin,tmax)
>>> tm
[[211813443995435008 211813445069176832]]
>>> from astropy.time import TimeDelta
>>> tm = mocpy.TimeMOC.from_time_ranges(tmin,tmax,delta_t=TimeDelta(1e-6,format='sec'))
>>> tm
[[211813444800000000 211813444800000001]]
```

We shouls have 2 intervals lasting 5 µs each, but we get only one. Internally, the code is converting input Time objects into jd (see [here](https://github.com/cds-astro/mocpy/blob/4e1c10725884f36d39535abf216b866e286e3993/mocpy/tmoc/tmoc.py#L80) for instance), and this destroys the microsecond resolution.

**Comment by fxpineau** [comment]
Fixed in v0.9.0 (current master branch) `TMOC.from_times` and `TMOC.from_time_ranges`.

**Comment by fxpineau** [comment]
Methods `times_to_microseconds` and `microseconds_to_times` added in `utils.py` and used in both 
`TimeMOC` and `STMOC` (and rust code updated accordingly to manipulate microseconds since jd=0 in u64 instead of jd in f64).
